src/NHibernate.Test/MappingByCode/ExplicitMappingTests/DynamicComponentMappingTests.cs

```diff
@@ -33,6 +33,137 @@ private class PersonWithGenericInfo
 			}
 		}
 
+		private class PersonWithDynamicInfo
+		{
+			public int Id { get; set; }
+			public dynamic Info { get; set; }
+		}
+
+		[Test]
+		public void WhenMapDynCompoByDictionaryThenMapItAndItsProperties()
+		{
+			//NH-3704
+			var mapper = new ModelMapper();
+			mapper.Class<Person>(
+				map =>
+				{
+					map.Id(x => x.Id, idmap => { });
+					map.Component(
+						x => x.Info,
+						new Dictionary<string, System.Type>
+							{ { "MyInt", typeof(int) }, { "MyDate", typeof(DateTime) } },
+						z => { });
```

Comment:
This PR isn't feature complete. Further configuration is not possible:
```suggestion
						z => z.Property("MyInt").Column("MyIntColumn"));
```

It throws: 
```
NHibernate.MappingException : Member not found. The member 'MyInt' does not exists in type System.Collections.Generic.Dictionary`2[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.Type, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]
```